### DISCUSSION TITLE
Bot support in Python

### DISCUSSION THREAD

**Discussion by jamestouri** [medplum/medplum#2175] [Ideas]
Given the recent rise in discussions about extending SDK support to other languages, it would be beneficial to document some of these thoughts. I propose that we leverage our bot services as a testing platform for integrating additional languages into our SDK. AWS Lambda simplifies the process of deploying Python scripts, making it an excellent starting point.

Examining a few demo-bots, we can see that initial interactions need not be with our Medplum client. Instead, we should begin by offering capabilities to engage with our FHIRTypes package.

Outlined below are the steps needed to introduce the first feature, which would enable users to interact with our FHIRTypes package:

1. Develop a Python SDK that supports the FHIRTypes
2. Configure AWS Lambda for the Python support
3. Extend the BotEditor.ts in the Medplum App to accommodate Python language capabilities, enabling the saving and deployment of bots
4. Establish CLI capabilities to create, save, and deploy bots using .py files

This approach offers a good foundation for creating Python bots and will facilitate the ongoing integration of support for cron jobs, scheduling, Medplum client capabilities, and more.  Would love any extra discussions on this.

**Comment by joekendal** [comment]
Golang would be nice
